sources/Valkey.Glide/Abstract/ConfigurationOptions.cs

```diff
@@ -476,9 +506,168 @@ private ConfigurationOptions DoParse(string configuration, bool ignoreUnknown =
+        tempAz = value;
+        // Don't validate immediately - wait until all parsing is complete
+    }
+
+    private void ValidateAndSetReadFrom()
+    {
+        if (tempReadFromStrategy.HasValue)
+        {
+            var strategy = tempReadFromStrategy.Value;
+
+            // Validate strategy and AZ combinations
+            switch (strategy)
+            {
+                case ReadFromStrategy.AzAffinity:
+                    if (string.IsNullOrWhiteSpace(tempAz))
+                    {
+                        throw new ArgumentException("Availability zone should be set when using AzAffinity strategy");
+                    }
+                    readFrom = new ReadFrom(strategy, tempAz);
+                    break;
+
+                case ReadFromStrategy.AzAffinityReplicasAndPrimary:
+                    if (string.IsNullOrWhiteSpace(tempAz))
+                    {
+                        throw new ArgumentException("Availability zone should be set when using AzAffinityReplicasAndPrimary strategy");
+                    }
+                    readFrom = new ReadFrom(strategy, tempAz);
+                    break;
+
+                case ReadFromStrategy.Primary:
+                    if (!string.IsNullOrWhiteSpace(tempAz))
+                    {
+                        throw new ArgumentException("Availability zone should not be set when using Primary strategy");
+                    }
+                    readFrom = new ReadFrom(strategy);
+                    break;
+
+                case ReadFromStrategy.PreferReplica:
+                    if (!string.IsNullOrWhiteSpace(tempAz))
+                    {
+                        throw new ArgumentException("Availability zone should not be set when using PreferReplica strategy");
+                    }
+                    readFrom = new ReadFrom(strategy);
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(tempReadFromStrategy), $"ReadFrom strategy '{strategy}' is not supported");
+            }
+        }
+    }
+
+    private static ReadFromStrategy ParseReadFromStrategy(string key, string value)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+        {
+            throw new ArgumentException($"Keyword '{key}' requires a ReadFrom strategy value; the value cannot be empty", key);
+        }
+
+        return value.ToLowerInvariant() switch
+        {
+            "primary" => ReadFromStrategy.Primary,
+            "preferreplica" => ReadFromStrategy.PreferReplica,
+            "azaffinity" => ReadFromStrategy.AzAffinity,
+            "azaffinityreplicasandprimary" => ReadFromStrategy.AzAffinityReplicasAndPrimary,
+            _ => throw new ArgumentException($"ReadFrom strategy '{value}' is not supported. Supported values are: Primary, PreferReplica, AzAffinity, AzAffinityReplicasAndPrimary", key)
+        };
+    }
+
+    private static void ValidateReadFromConfiguration(ReadFrom readFromConfig)
```

Comment:
This method repeats `ValidateAndSetReadFrom`. Can you remove duplicates or re-use the code?